### PR TITLE
feat(input): add emoji

### PR DESCRIPTION
**Pull request by lyjeileen**
## Change
- Add emoji to BaseInput
- Add tests and stories for it
- Try to fix the failing test for pdfViewer

Note: I added the emoji icon next to the 'speech to text' icon for now. Since the icon button color should change on Input's focus state, I added an optional color props to pass the correct color to the emoji button.
Had this error before: `Failed to execute 'transaction' on 'IDBDatabase': The database connection is closing.` and it's fixed by adding `database.close()`. Reference: https://github.com/nolanlawson/emoji-picker-element/issues/220#issuecomment-906082810
### Before
Screenshot 2024-06-26 at 5 29 11 PM

### After
Screenshot 2024-06-29 at 12 08 19 AM

Screenshot 2024-06-29 at 12 08 10 AM

Screenshot 2024-06-28 at 5 55 47 PM
Screenshot 2024-06-28 at 5 57 36 PM

## Video
Jun-27-2024 18-18-23

### CODE REVIEW

**Comment by kaseyvee** [comment]
The emoji pop up uses my system settings theme at the moment (dark mode) no matter which theme is set on storybook. Not sure if that's something you want to address in this pr or the styling follow up.
Screenshot 2024-06-28 at 10 58 42 AM

However the search popup remains consistent with the storybook theme.
Screenshot 2024-06-28 at 10 58 48 AM

**Comment by kaseyvee** [comment]
Would be nice if the predictive popover was accessible through keyboard. Could maybe do in a follow up

**Comment by lyjeileen** [comment]
> Would be nice if the predictive popover was accessible through keyboard. Could maybe do in a follow up

Maybe we could handle this in a follow up? I will fix the theme issue

File: src/components/input/baseInput/baseInput.tsx [hunk 1]
```diff
@@ -35,46 +68,80 @@ function BaseInputElement(
...
-        )}
-      </InputAdornment>
-    ),
+  const database = new Database()
+
+  function searchEmojis(query: string) {
+    database
+      .getEmojiBySearchQuery(query)
+      .then((results) => {
+        const resultLimit = 5
```
Comment (Shiti) [inline src/components/input/baseInput/baseInput.tsx:115]: could this be configurable ? We could default it to 5

**Review by Shiti — DISMISSED** [review]

**Review by kaseyvee — APPROVED** [review]

File: src/components/input/emoji/emoji.tsx [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+    }
+  }
+
+  function handleEmojiClick(event: EmojiClickEvent) {
+    event.detail.unicode && props.onEmojiClick(event.detail.unicode)
+    const pickerElement = document.querySelector(
+      'emoji-picker'
+    ) as EmojiPickerElement
+    pickerElement?.database.close().catch((error) => {
+      console.error('Failed to close database:', error)
```
Comment (Shiti) [inline src/components/input/emoji/emoji.tsx:52]: nit
```suggestion
      console.error('EmojiPicker: Failed to close database:', error)
```

**Review by Shiti — APPROVED** [review]

**Pull request merged**